proxy/tests/discovery.rs

```diff
@@ -1,95 +1,129 @@
+        }
+
+        #[test]
+        fn outbound_uses_orig_dst_if_not_local_svc() {
+            let _ = env_logger::try_init();
+
+            let srv = $make_server()
+                .route("/", "hello")
+                .route("/bye", "bye")
+                .run();
+            let ctrl = controller::new()
+                // no controller rule for srv
+                .run();
+            let proxy = proxy::new()
+                .controller(ctrl)
+                // set outbound orig_dst to srv
+                .outbound(srv)
+                .run();
+            let client = $make_client(proxy.outbound, "versioncheck.conduit.io");
+
+            assert_eq!(client.get("/"), "hello");
+            assert_eq!(client.get("/bye"), "bye");
+        }
+
+        #[test]
+        fn outbound_asks_controller_without_orig_dst() {
+            let _ = env_logger::try_init();
+
+            let srv = $make_server()
+                .route("/", "hello")
+                .route("/bye", "bye")
+                .run();
+            let ctrl = controller::new()
+                .destination("disco.test.svc.cluster.local", srv.addr)
+                .run();
+            let proxy = proxy::new()
+                .controller(ctrl)
+                // don't set srv as outbound(), so that SO_ORIGINAL_DST isn't
+                // used as a backup
+                .run();
+            let client = $make_client(proxy.outbound, "disco.test.svc.cluster.local");
+
+            assert_eq!(client.get("/"), "hello");
+            assert_eq!(client.get("/bye"), "bye");
+        }
+    }
 }
 
-#[test]
-fn outbound_reconnects_if_controller_stream_ends() {
-    let _ = env_logger::try_init();
+mod http2 {
+    use super::support::*;
+    use std::thread;
 
-    let srv = server::new().route("/recon", "nect").run();
-    let ctrl = controller::new()
-        .destination_close("disco.test.svc.cluster.local")
-        .destination("disco.test.svc.cluster.local", srv.addr)
-        .run();
-    let proxy = proxy::new().controller(ctrl).outbound(srv).run();
-    let client = client::new(proxy.outbound, "disco.test.svc.cluster.local");
+    generate_tests! { server: server::new, client: client::new }
 
-    assert_eq!(client.get("/recon"), "nect");
 }
 
-#[test]
-fn outbound_http1_asks_controller_about_host() {
-    let _ = env_logger::try_init();
+mod http1 {
+    use super::support::*;
+    use std::thread;
 
-    let srv = server::http1()
-        .route("/", "hello")
-        .route("/bye", "bye")
-        .run();
-    let ctrl = controller::new()
-        .destination("disco.test.svc.cluster.local", srv.addr)
-        .run();
-    let proxy = proxy::new()
-        .controller(ctrl)
-        // don't set srv as outbound(), so that SO_ORIGINAL_DST isn't
-        // used as a backup
-        .run();
-    let client = client::http1(proxy.outbound, "disco.test.svc.cluster.local");
+    generate_tests! { server: server::http1, client: client::http1 }
```

Comment:
We should also have a case that uses `client::http1_absolute_uris` too.

Reply:
Done in db2cec50b75d9950b9dfe4aa932c539c7a14e61d!